eth/stagedsync/stage_headers.go

```diff
@@ -188,18 +201,25 @@ func HeadersPOS(
 			LatestValidHash: headerHash,
 		}
 
-		if err := headerInserter.FeedHeaderPoS(tx, &header, headerHash); err != nil {
+		if err := headerInserter.FeedHeaderPoS(tx, header, headerHash); err != nil {
+			return err
+		}
+		// We can insert raw bodies immediately and skip bodies
+		if err := rawdb.WriteRawBody(tx, headerHash, headerNumber, payloadMessage.Body); err != nil {
```

Comment:
And we also need to write the block hash.

Reply:
block hash is another stage not in bodies

Reply:
there is a blockhashes stage for this

Reply:
Ah, OK. It's a bit inconsistent though: you let the 2nd stage (BlockHashes) to run its natural course, while you save the block and manually set the progress on the 3rd stage (Bodies). Better at least put a comment in the code about this inconsistency.

Reply:
added a TODO